src/Symfony/Component/Messenger/EventListener/SendFailedMessageForRetryListener.php

```diff
@@ -70,7 +71,18 @@ public function onMessageFailed(WorkerMessageFailedEvent $event)
             $delay = $retryStrategy->getWaitingTime($envelope, $throwable);
 
             if (null !== $this->logger) {
-                $this->logger->error('Error thrown while handling message {class}. Sending for retry #{retryCount} using {delay} ms delay. Error: "{error}"', $context + ['retryCount' => $retryCount, 'delay' => $delay, 'error' => $throwable->getMessage(), 'exception' => $throwable]);
+                $logLevel = LogLevel::ERROR;
+                if ($throwable instanceof RecoverableExceptionInterface) {
+                    $logLevel = LogLevel::WARNING;
+                } elseif ($throwable instanceof HandlerFailedException) {
+                    foreach ($throwable->getNestedExceptions() as $nestedException) {
+                        if ($nestedException instanceof RecoverableExceptionInterface) {
+                            $logLevel = LogLevel::WARNING;
```

Comment:
It should only be changed to warning if all of the handlers threw recoverable exceptions, no?

Reply:
I copied the behavior from the shouldRetry method, So I think this is fine like this. But I'm not sure indeed

Reply:
Agreed: one handler throwing a `RecoverableExceptionInterface` is enougth to retry the message.